client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js

```diff
@@ -376,6 +376,26 @@ angular.module('managerApp')
             self.states.hasOldFlavors = true;
           }
         });
+        // Remove flavors if OS has restricted
+        const restrictedFlavors = _.get(self.vmInEdition, 'image.flavorType') || [];
+        if (restrictedFlavors.length > 0) {
+          self.displayData.categories = _.filter(
+            self.displayData.categories,
+            (category) => {
+              // eslint-disable-next-line no-param-reassign
+              category.flavors = _.filter(
+                category.flavors,
+                flavor => _.indexOf(restrictedFlavors, flavor.shortType) > -1,
+              );
+              return category.flavors.length > 0;
+            },
+          );
```

Comment:
```suggestion
           self.displayData.categories = self.displayData.categories.filter(category => 
            category.flavors.some(flavor => restrictedFlavors.includes(flavor.shortType)))
```

Reply:
this suggestion does not work for my use case